Gas fee optimization

Yingjie-Zhang-dev opened the pull request:
Implement gas fee optimization. 
Both files have been tested to work properly.

Yingjie-Zhang-dev committed:
Gas fee optimization

Implement gas fee optimization. 
Both files have been tested to work properly.

finnformica commented on CauseContract.sol:
```diff
@@ -34,7 +37,7 @@ contract CauseContract {
     mapping(address => uint256) public donorTotals;
     
-    mapping(address => bool) public addressDonated;
+    // mapping(address => bool) public addressDonated;
```
what is this comment for?

Yingjie-Zhang-dev replied:
I kept the original code in the form of comment for comparison with the changed code.
Here I changed mapping to array for gas saving. Calm, I just don't want to cause any confusion

finnformica commented on CauseContract.sol:
```diff
@@ -67,9 +70,14 @@ contract CauseContract {
-        contractAddress = payable(address(this));
+
+        //  directly use address(this) whenever needed can save some storage and gas
+        // contractAddress = payable(address(this));
```
delete these comments pls

Yingjie-Zhang-dev replied:
No problem.

Yingjie-Zhang-dev committed:
Delete some confusing comments

finnformica commented on CauseFactory.sol:
```diff
@@ -33,7 +33,8 @@ contract CauseFactory {
         return infos;
     }
 
-    function cfRetrieveIds() public view returns (string[] memory) {
```
Could you leave this function in? I've taken a look at the docs and apprently you can only access one value of an array or mapping at a time, so returning the entire array is more efficient...

Yingjie-Zhang-dev replied:
Okay🫡

finnformica replied:
Could you uncomment the code pls and then I'll approve the commit

Yingjie-Zhang-dev replied:
Sorry! I thought I had uploaded the ucommented file 2 days ago, but it seems like the file didn't successfully uploaded at that time. 
It should works now!

Yingjie-Zhang-dev committed:
active cfRetrieveIds()

Yingjie-Zhang-dev committed:
Active cfRetrieveIds()

Yingjie-Zhang-dev committed:
Removing the gas fee calculation and storage

1. Remove gasUsed and transactionFee from the Transaction struct
2. Remove gas fee calculation from the donate() function
3. Remove gas fee calculation from the withdraw() function
By removing the gas fee calculation and storage, we can save gas and solve the problem of gas used calculation at the same time.
